File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/decorator/uielement/BorderDecorator.java
// BorderDecorator.java

package com.bigboxcode.designpattern.decorator.uielement;

public class BorderDecorator extends UIDecorator {
    protected BorderDecorator(UIElement uiElement) {
        super(uiElement);
    }

    @Override
    public void draw() {
        // Can perform any additional task anywhere in the method

        super.draw();

        // Write code to add border to the element

        System.out.println("Adding Border to the element");
    }
}

File: design-patterns/ts/decorator/ui-element/border-decorator.ts
// border-decorator.ts

import UIDecorator from "./ui-decorator";
import UIElement from "./ui-element";

class BorderDecorator extends UIDecorator {
    constructor(uiElement: UIElement) {
        super(uiElement);
    }

    draw(): void {
        super.draw();
        console.log("Adding Border to the element");
    }
}

export default BorderDecorator;